# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: b326d33f98cb
# Full Hash: b326d33f98cbc1d08d8ca10609ae24e8c852767c
# Author: Andreas Farre <farre@mozilla.com>
# Date: 2020-08-01 09:48:10
# Regressor Bug: 1613431
# File Overlap Count: 1
# Description:
#   Bug 1613431 - Part 2: Ignore synced setters return value. r=nika
#   
#   Depends on D83645
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83646
# ==============================================================================

diff -r 4d29677fb419 -r b326d33f98cb docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Fri Jul 31 13:36:45 2020 +0000
+++ b/docshell/base/BrowsingContext.cpp	Fri Jul 31 13:37:00 2020 +0000
@@ -1159,11 +1159,13 @@
 }
 
 void BrowsingContext::NotifyUserGestureActivation() {
-  SetUserActivationState(UserActivation::State::FullActivated);
+  // Return value of setting synced field should be checked. See bug 1656492.
+  Unused << SetUserActivationState(UserActivation::State::FullActivated);
 }
 
 void BrowsingContext::NotifyResetUserGestureActivation() {
-  SetUserActivationState(UserActivation::State::None);
+  // Return value of setting synced field should be checked. See bug 1656492.
+  Unused << SetUserActivationState(UserActivation::State::None);
 }
 
 bool BrowsingContext::HasBeenUserGestureActivated() {
@@ -1201,7 +1203,9 @@
   top->PreOrderWalk([&](BrowsingContext* aContext) {
     if (aContext->GetUserActivationState() ==
         UserActivation::State::FullActivated) {
-      aContext->SetUserActivationState(UserActivation::State::HasBeenActivated);
+      // Updating user activation state on a discarded context has no effect.
+      Unused << aContext->SetUserActivationState(
+          UserActivation::State::HasBeenActivated);
     }
   });
 
@@ -2222,7 +2226,8 @@
   if (XRE_IsParentProcess()) {
     PreOrderWalk([&](BrowsingContext* aContext) {
       if (aContext != this) {
-        aContext->SetDefaultLoadFlags(loadFlags);
+        // Setting load flags on a discarded context has no effect.
+        Unused << aContext->SetDefaultLoadFlags(loadFlags);
       }
     });
   }
@@ -2428,7 +2433,8 @@
     }
 
     for (BrowsingContext* child : Children()) {
-      child->SetTextZoom(GetTextZoom());
+      // Setting text zoom on a discarded context has no effect.
+      Unused << child->SetTextZoom(GetTextZoom());
     }
   }
 
@@ -2458,7 +2464,8 @@
     }
 
     for (BrowsingContext* child : Children()) {
-      child->SetFullZoom(GetFullZoom());
+      // Setting full zoom on a discarded context has no effect.
+      Unused << child->SetFullZoom(GetFullZoom());
     }
   }
 